src/main/logutils.ts

```diff
@@ -665,46 +669,49 @@ const forceStopRecording = () => {
 }
 
 /**
- * checkWoWProcess
- * @returns {[boolean, boolean]} retailRunning, classicRunning
+ * Check Windows task list and find any WoW process.
  */
-const checkWoWProcess = async (): Promise<[boolean, boolean]> => {
-    let retailRunning = false;
-    let classicRunning = false;
-
+const checkWoWProcess = async (): Promise<IWoWProcessResult[]> => {
+    const wowProcessRx = new RegExp(/(wow(T|B|classic)?)\.exe/, 'i');
     const taskList = await tasklist();
 
-    taskList.forEach((process: any) => {
-        if (process.imageName === "Wow.exe") {
-            retailRunning = true;
-        } else if (process.imageName === "WowClassic.exe") {
-            classicRunning = true;
-        }
-    });
-
-    return [retailRunning, classicRunning]
+    return taskList
+        // Map all processes found to check if they match `wowProcessRx`
+        .map((p: any) => p.imageName.match(wowProcessRx))
+        // Remove those that result in `null` (didn't match)
+        .filter((p: any) => p)
+        // Return an object suitable for `IWoWProcessResult`
+        .map((match: any): IWoWProcessResult => ({
+            exe: match[0],
+            flavour: wowExecutableFlavours[match[1].toLowerCase()]
+        }))
+    ;
 }
 
 /**
  * pollWoWProcessLogic
  */
-const pollWoWProcessLogic = async (startup: boolean) => {
-    const [retailFound, classicFound] = await checkWoWProcess();
-    const retailProcessChanged = (retailFound !== isRetailRunning);
-    // TODO classic support
-    const classicProcessChanged = (classicFound !== isClassicRunning);
-    const processChanged = (retailProcessChanged || classicProcessChanged);
-    if (!retailProcessChanged && !startup) return;
-    (retailFound) ? wowProcessStarted() : wowProcessStopped();
+const pollWoWProcessLogic = async () => {
+    const wowProcesses = await checkWoWProcess();
+    const wowProcess = wowProcesses.pop();
```

Comment:
So we'll only spot one, but we don't do any custom logic now so this is fine? If all 4 are running we'll spot one. No objection just confirming.

Reply:
Precisely!